app/serializers/user_topic_bookmark_serializer.rb

```diff
@@ -28,28 +27,7 @@ def raw
   end
 
   def cooked
-    @cooked ||= \
-      if last_read_post_number.present?
-        for_topic_cooked_post
-      else
-        first_post.cooked
-      end
-  end
-
-  def for_topic_cooked_post
-    post_number = [last_read_post_number + 1, highest_post_number].min
-    sorted_regular_posts = topic.posts.sort_by(&:post_number).select do |post|
-      post.post_type == Post.types[:regular]
-    end
-    first_unread_post = sorted_regular_posts.find do |post|
-      post.post_number >= post_number
-    end
-
-    # if first_unread_cooked is blank this likely means that the last
-    # read post was either deleted or is a small action post.
-    # in this case we should just get the last regular post and
-    # use that for the cooked value so we have something to show
-    (first_unread_post || sorted_regular_posts.last).cooked
+    first_post.cooked
```

Comment:
I think we need to switch from preloading `posts`, to preloading the `first_post` relation here:

https://github.com/discourse/discourse/blob/72ea00d0cc69e49017ecb363f2dc1a39aa2dea14/app/services/topic_bookmarkable.rb#L14-L16

So that would become:

```ruby
def self.preload_associations
  [:tags, { first_post: :user }]
end
```

Reply:
Ah good catch! Thanks, will change :+1: